docs/design/net-aware.md

```diff
@@ -0,0 +1,64 @@
+# Net Topology Aware Plugin
+
+## Backgrounds
+
+Usually, a kubernetes cluster has many nodes, and those nodes are in different idc, chassis, and even different switches.
+Data transformations across different idc, chassis and switches has different performance. Some latency sensitive workloads are need to run in same idc, even in same topology devices, such as chassis and switch.
+
+## Motivation
+
+We target to make scheduler net-topology aware so as to achieve the following:
+
+- best effort to schedule same job to same topology devices, such as same idc, chassis or switch.
+
+## Goals
+
+- Support single-key topology configuration, try to schedule job's all tasks to nodes which have same value with that key
+- Support multiple-key topology policies, the key at front get higher score
+
+## Non-Goals
+
+- Not to find the global solutions among nodes with all values of that key
+
+## Design Action
+
+### Pod scheduling process
+
+1. when the first task of a job is allocated to a node, record the node information in the plugin
+2. when scheduling other tasks, a node with same key-value as the target node, get a higher score. Otherwise, get a zero score.
+3. If a node has multiple keys same as the configured list, the first key matching the configured keys has higher score
+
+```go
+nodeOrderFn := func(task *api.TaskInfo, node *api.NodeInfo) (float64, error){
```

Comment:
From the original user demand, job should occupy nodes execulicm the original user demand, job should occupy nodes exclusively, so only nodeorder func seems cannot satisfy the original use case.

Reply:
You mean  a predicateFn is need?